leetcode_arrays.js

```diff
@@ -0,0 +1,107 @@
+/***************** Max Consecutive Ones */
+/**
+ * @param {number[]} nums
+ * @return {number}
+ */
+var findMaxConsecutiveOnes = function(nums) {
+    var maxcon = 0;
+    var currentcon = 0;
+    for(i=0;i<nums.length;++i){
+        num = nums[i];
+        if(num == 1){
+            currentcon++;
+        } else {
+            currentcon = 0;
+        }
+        if(currentcon > maxcon)
+            maxcon = currentcon;
+    }
+    return maxcon;
+};
+
+/***************** Find Numbers With Even Digits */
+/**
+ * @param {number[]} nums
+ * @return {number}
+ */
+var findNumbers = function(nums) {
+    edn = 0;
+    for(let i=0;i<nums.length;i++){
+        num = nums[i];
+        digitsnum = String(num).length;
+        if(digitsnum % 2 == 0)
+            edn += 1;
+    }
+    return edn;
+};
+
+/***************** Squares of a Sorted Array */
+/**
+ * @param {number[]} nums
+ * @return {number[]}
+ */
+var sortedSquares = function(nums) {
+    var squaresarray = [];
+    for(let i=0;i<nums.length;i++){
+        num = nums[i];
+        squaresarray[i] = num*num;
+    }
+    squaresarray.sort(function(a,b){
+        return a - b;
+    });
+    return squaresarray;
+};
+
+/***************** Duplicate Zeros */
+/**
+ * @param {number[]} arr
+ * @return {void} Do not return anything, modify arr in-place instead.
+ */
+
+var shiftArray = function(arr, stindx){
+    for(let i=arr.length-1;i>stindx;i--){
+        arr[i] = arr[i-1];
+    }
+}
+
+var duplicateZeros = function(arr) {
+    for(let i=0;i<arr.length;i++){
+        num = arr[i];
+        if(num == 0){
+            shiftArray(arr,i);
+            i++;
```

Comment:
this is O(n^2) right? since shiftarray makes a pass through the whole rest of the array.  there should be a O(n) way to solve this, maybe see what you can do if you go through the array once to count how many zeros there are?